src/tools/perf/lib/libperf.c

```diff
@@ -1028,6 +1041,35 @@ ucp_perf_test_rkey_unpack(ucx_perf_thread_context_t *thread,
     return UCS_OK;
 }
 
+static ucs_status_t ucp_perf_test_create_self_rkey(ucx_perf_context_t *perf,
+                                                   ucp_mem_h memh,
+                                                   ucp_rkey_h *rkey)
+{
+    void *rkey_buffer = NULL;
+    size_t rkey_size  = 0;
+    ucs_status_t status;
+
+    status = ucp_rkey_pack(perf->ucp.context, memh, &rkey_buffer, &rkey_size);
+    if (status != UCS_OK) {
+        if (perf->params.flags & UCX_PERF_TEST_FLAG_VERBOSE) {
+            ucs_fatal("ucp_rkey_pack() failed: %s", ucs_status_string(status));
+        }
+        goto out;
+    }
+
+    status = ucp_ep_rkey_unpack(perf->ucp.self_ep, rkey_buffer, rkey);
+    if (status != UCS_OK) {
+        if (perf->params.flags & UCX_PERF_TEST_FLAG_VERBOSE) {
+            ucs_fatal("ucp_rkey_unpack() failed: %s",
+                      ucs_status_string(status));
+        }
+    }
```

Comment:
Maybe add a macro. And use it here and in `ucp_perf_test_rkey_pack`, `ucp_perf_test_rkey_unpack`.